feat: compute genesis root

gnapoli23 opened the pull request:
### Describe the feature
Genesis state root is not computed, but is a mandatory field inside `Genesis` struct.
This should be computed rather then be required, using `genesis_state_root` fn, to be in line with Geth behaviour.
It also includes tests for `init_genesis`.

### Additional context
Fixes #896

gnapoli23 commented:
Atm, i wrote 3 basic tests for `init_gensis`. While first two passes, the third one `success_init_genesis_sepolia` doesn't.

```
computed_genesis: '0x2f7e7be7b9b001d477a981dcd9a4ffe20ff8d90f92e9b0f366cdc32efc789aab',
GENESIS_SEPOLIA: '0x25a5cc106eea7138acab33231d7160d69cb777ee0c2c553fcddf5138993e6dd9'
```

Rjected commented:
Ah, this is due to a current bug in how we set up forks for sepolia - will fix

Rjected commented on crates/staged-sync/src/utils/init.rs:
```diff
@@ -68,3 +68,50 @@ pub fn init_genesis<DB: Database>(db: Arc<DB>, chain: ChainSpec) -> Result<H256,
+        let db = create_test_rw_db::<WriteMap>();
+
+        let mut chain_spec = MAINNET.clone();
+        chain_spec.genesis_hash = Header::default().hash_slow();
```
the genesis hash in the `chain_spec` is not being used in `init_genesis` so I don't think setting it in these tests is necessary

Rjected reviewed (commented):
looks great so far, just an initial comment on the tests, will take a look again pending rebase after #1021 is in

gnapoli23 commented:
@Rjected 
Ok I updated tests removing those `genesis_hash` setting and rebased on top of latest commit. The problem about sepolia genesis init test seems solved.

Rjected reviewed (approved):
looks like there are some conflicts that need to be fixed, otherwise this looks great! tysm!

gnapoli23 commented:
> looks like there are some conflicts that need to be fixed, otherwise this looks great! tysm!

@Rjected Ty! Rebased, and fixed conflicts 👍🏻

Rjected commented:
nice! gg
